### PR TITLE
update deprecated Array constructor

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

**Comment by musm** [comment]
master failures might be related to https://github.com/JuliaLang/julia/pull/20084 altough I have not actively looked into this

File: src/BinDeps.jl [hunk 1]
```diff
@@ -194,7 +194,7 @@ end
 
 type Choices <: BuildStep
     choices::Vector{Choice}
-    Choices() = new(Array(Choice,0))
+    Choices() = new(Vector{Choice}())
```
Comment (musm) [inline src/BinDeps.jl]: `Choice[]` is also possible here and below, not sure which you prefer?

Comment (tkelman) [inline src/BinDeps.jl]: there's some discussion that `Vector{T}()` might get deprecated - hasn't happened yet, but evidently the zero arg form isn't loved by all

**Comment by musm** [comment]
I feel like a lot of the inner constructors here are unnecessary

File: src/dependencies.jl [hunk 2]
```diff
@@ -239,7 +239,7 @@ immutable SystemPaths <: DependencyProvider; end
 
 show(io::IO, ::SystemPaths) = print(io,"System Paths")
 
-using URIParser, Compat
```
Comment (musm) [inline src/dependencies.jl]: module already has a `using Compat` so it is not needed to repeat it.

**Comment by tkelman** [comment]
Thanks

**Pull request merged**